hackathon_site/hardware/test_api.py

```diff
@@ -607,6 +608,58 @@ def test_search_filter(self):
         self.assertCountEqual(returned_ids, [self.order_3.id, self.order_4.id])
 
 
+class IncidentListViewPostTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.team = Team.objects.create()
+        self.order = Order.objects.create(status="Cart", team=self.team)
+        self.permissions = Permission.objects.filter(
+            content_type__app_label="hardware", codename="add_incident"
+        )
+
+        self.hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.order_item = OrderItem.objects.create(
+            order=self.order, hardware=self.hardware,
+        )
+
+        self.incident = Incident.objects.create(
+            state="Broken",
+            description="Description",
+            order_item=self.order_item,
+            time_occurred=datetime(2022, 8, 8, tzinfo=settings.TZ_INFO),
+        )
+
+        self.request_data = IncidentCreateSerializer(self.incident).data
```

Comment:
I think it would actually be better to hard code the json we'd be submitted as part of the test rather than using the serializer to generate it, since that's how the frontend is going to use it after all. Plus then it would be easier to notice the nested field that is in here I think.

Reply:
yes will do